### PR TITLE
Fix for the new glyph cache

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: engine/render/src/render/font_renderer.cpp [hunk 1]
```diff
@@ -313,7 +313,7 @@ namespace dmRender
         font_map->m_CacheCellPadding = params.m_CacheCellPadding;
         font_map->m_CacheChannels = params.m_GlyphChannels;
 
-        SetupCache(font_map, font_map->m_CacheWidth, font_map->m_CacheWidth,
+        SetupCache(font_map, font_map->m_CacheWidth, font_map->m_CacheHeight,
```
Comment (JCash) [inline engine/render/src/render/font_renderer.cpp:316]: Wrong size.

File: engine/render/src/render/font_renderer.cpp [hunk 2]
```diff
@@ -781,16 +781,16 @@ namespace dmRender
         dmGraphics::SetTexture(font_map->m_Texture, tex_params);
     }
 
-    static void AddGlyphToCache(HFontMap font_map, uint32_t frame, uint32_t c, dmRender::FontGlyph* g, int32_t g_offset_y)
+    static void AddGlyphToCache(HFontMap font_map, uint32_t frame, dmRender::FontGlyph* g, int32_t g_offset_y)
```
Comment (JCash) [inline engine/render/src/render/font_renderer.cpp:784]: Let's not confuse things with the same variable in two places.

File: engine/render/src/render/font_renderer.cpp [hunk 3]
```diff
@@ -898,6 +897,9 @@ namespace dmRender
                         continue;
                     }
 
+                    // If we get here, then it may be that c != glyph->m_Character
+                    c = g->m_Character;
```
Comment (JCash) [inline engine/render/src/render/font_renderer.cpp:901]: Old behavior:
If the glyph doesn't exist, it checks if 126 `~` exists

So the fix is to simply use the codepoint of the glyph we retrieved (and not the one we originally  requested)

**Review by AGulev — APPROVED** [review]
🔥

**Pull request merged**